### DISCUSSION TITLE
LSP inlay hints toggle doesn't work

### DISCUSSION THREAD

**Discussion by hsheikhali1** [LazyVim/LazyVim#967] [Q&A]
I'm on neovim nightly, but for some reason, inlay hints don't work, and neither does the toggle option baked into which key - am I missing something??

**Comment by folke** [comment]
Yes, you need to enable inlay hints for the lsp servers that provide them

**Reply by hsheikhali1** [reply]
Do you have an example of how to do that?

**Reply by binary4cat** [reply]
I guess this may need to be configured. Since my nvim version does not have that (the stable version I use), I have not verified whether it is correct.
```lua
{
    "neovim/nvim-lspconfig",
    opts = function(_, opts)
      opts.inlay_hints.enabled = true
    end,
  },
```

**Reply by hsheikhali1** [reply]
Nope - doesn't work.

**Reply by binary4cat** [reply]
It may be necessary to configure inlay hints separately for each language LSP. For example, gopls of golang language can use [this configuration document](https://github.com/golang/tools/blob/master/gopls/doc/inlayHints.md). You need to find the corresponding language LSP document to configure.

**Reply by hsheikhali1** [reply]
Yeah like @folke said I had to enable it first for the server I want and boom it worked :) However, `Shift + K` no longer works lol and `Glance` is kind of broken.
